File: backend/src/main/scala/kpn/api/common/search/RouteListItem.scala
package kpn.api.common.search

import kpn.api.common.Bounds

case class RouteListItem(
  id: Long,
  name: String,
  distance: Long,
  symbol: Option[String],
  bounds: Option[Bounds],
  routeIds: Seq[Long],
)

File: backend/api/src/main/java/kpn/api/common/search/RouteListItem.java
package kpn.api.common.search;

import kpn.api.common.Bounds;

import java.util.Optional;
import com.google.common.collect.ImmutableList;

public record RouteListItem(
  Long id,
  String name,
  Long distance,
  Optional<String> symbol,
  Optional<Bounds> bounds,
  ImmutableList<Long> routeIds
) {}
